### ISSUE TITLE
feat(EmptyState): allow props to be functions for maximum flexibility

### ISSUE THREAD

**Issue by devfreddy** [newrelic/nr1-community#60]
## Summary

The current prop list for `<EmptyState>` looks like:
```jsx
    heading: PropTypes.string,
    description: PropTypes.string,
    buttonText: PropTypes.string,
    buttonOnClick: PropTypes.func,
    buttonUrl: PropTypes.string,
    className: PropTypes.string,
    featuredImage: PropTypes.string,
    footer: PropTypes.func
```

## Desired Behavior

More flexible options for customization. For example:

```jsx
    <EmptyState description={() => {
        return (
            <React.Fragment>
                <p>Paragraph 1</p>
                <br/>
                <p>Paragraph 2</p>
            </React.fragment>
        )
      }}
    </EmptyState>
```
## Possible Solution

Let's investigate and make a change to allow for any of them to be a function OR a string, like this:

```jsx
    heading: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
    description: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
    buttonText: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
    buttonOnClick: PropTypes.func,
    buttonUrl: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
    className: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
    featuredImage: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
    footer: PropTypes.func
```

## Additional context

[TIP]:  # ( Why does this feature matter to you? What unique circumstances do you have? )

**Comment by danielgolden** [comment]
@devfreddy Do you have a use case in mind for providing a function to `buttonUrl` and `className`? Wouldn't we always want them to be a string?

**Comment by tangollama** [comment]
Closed by #74
